core/variant/variant_call.cpp

```diff
@@ -657,22 +657,37 @@ static _FORCE_INLINE_ void vc_ptrcall(void (*method)(T *, P...), void *p_base, c
 		}                                                                                                                                                         \
 	};
 
-#define VARCALL_PACKED_GETTER(m_packed_type, m_return_type)                                       \
-	static m_return_type func_##m_packed_type##_get(m_packed_type *p_instance, int64_t p_index) { \
-		return p_instance->get(p_index);                                                          \
+#define VARCALL_PACKED_GETTER_SETTER(m_packed_type, m_type)                                                     \
+	static m_type func_##m_packed_type##_get(m_packed_type *p_instance, int64_t p_index) {                      \
+		ERR_FAIL_INDEX_V(p_index, p_instance->size(), m_type());                                                \
+		return p_instance->get(p_index);                                                                        \
+	}                                                                                                           \
+	static void func_##m_packed_type##_set(m_packed_type *p_instance, int64_t p_index, const m_type &p_value) { \
+		ERR_FAIL_INDEX(p_index, p_instance->size());                                                            \
+		p_instance->set(p_index, p_value);                                                                      \
 	}
 
 struct _VariantCall {
-	VARCALL_PACKED_GETTER(PackedByteArray, uint8_t)
-	VARCALL_PACKED_GETTER(PackedColorArray, Color)
-	VARCALL_PACKED_GETTER(PackedFloat32Array, float)
-	VARCALL_PACKED_GETTER(PackedFloat64Array, double)
-	VARCALL_PACKED_GETTER(PackedInt32Array, int32_t)
-	VARCALL_PACKED_GETTER(PackedInt64Array, int64_t)
-	VARCALL_PACKED_GETTER(PackedStringArray, String)
-	VARCALL_PACKED_GETTER(PackedVector2Array, Vector2)
-	VARCALL_PACKED_GETTER(PackedVector3Array, Vector3)
-	VARCALL_PACKED_GETTER(PackedVector4Array, Vector4)
+	VARCALL_PACKED_GETTER_SETTER(PackedByteArray, uint8_t)
+	VARCALL_PACKED_GETTER_SETTER(PackedColorArray, Color)
+	VARCALL_PACKED_GETTER_SETTER(PackedFloat32Array, float)
+	VARCALL_PACKED_GETTER_SETTER(PackedFloat64Array, double)
+	VARCALL_PACKED_GETTER_SETTER(PackedInt32Array, int32_t)
+	VARCALL_PACKED_GETTER_SETTER(PackedInt64Array, int64_t)
+	VARCALL_PACKED_GETTER_SETTER(PackedStringArray, String)
+	VARCALL_PACKED_GETTER_SETTER(PackedVector2Array, Vector2)
+	VARCALL_PACKED_GETTER_SETTER(PackedVector3Array, Vector3)
+	VARCALL_PACKED_GETTER_SETTER(PackedVector4Array, Vector4)
+
+	static Variant func_Array_get(Array *p_instance, int64_t p_index) {
+		ERR_FAIL_INDEX_V(p_index, p_instance->size(), Variant());
+		return p_instance->get(p_index);
+	}
+
+	static void func_Array_set(Array *p_instance, int64_t p_index, const Variant &p_value) {
+		ERR_FAIL_INDEX(p_index, p_instance->size());
+		p_instance->set(p_index, p_value);
+	}
```

Comment:
Seems like this could be:
```suggestion
	VARCALL_PACKED_GETTER_SETTER(Array, Variant)
```
In which case maybe the macro should be renamed to `VARCALL_ARRAY_GETTER_SETTER` since it would not just be about packed arrays.

Reply:
Done.